app/src/main/java/ch/epfl/sdp/utils/RandomGenerator.java

```diff
@@ -132,12 +132,12 @@ public Enemy randomEnemy() {
          GeoPoint g5 = randomGeoPoint();
          Player p5 = new Player(g5.getLongitude(), g5.getLatitude(), rand.nextDouble()+50, randomString(10), randomEmail());
 
-         PlayerManager playerManager = new PlayerManager();
-         PlayerManager.addPlayer(p1);
-         PlayerManager.addPlayer(p2);
-         PlayerManager.addPlayer(p3);
-         PlayerManager.addPlayer(p4);
-         PlayerManager.addPlayer(p5);
+         PlayerManager playerManager = PlayerManager.getInstance();
+         playerManager.addPlayer(p1);
```

Comment:
It might not be necessary, but maybe you can create an addPlayers(Player...p) , this function takes a variable amount of args, and then you can just do playerManager.addPlayers(p1,p2,p3,p4,p5) (instead of having a new line for each player). Also, by creating a new method and testing it, you might slightly increase the test coverage.

Reply:
Thanks for advice, will try